xyz-hub-service/src/main/java/com/here/xyz/hub/auth/SpaceAuthorization.java

```diff
@@ -176,77 +179,93 @@ public static void authorizeModifyOp(ConditionalOperation task, Callback<Conditi
-        task.canReadConnectorsProperties = tokenRights != null && tokenRights.matches(connectorsRights);
-        if (!task.canReadConnectorsProperties) {
-          throw new HttpException(FORBIDDEN, getForbiddenMessage(connectorsRights, tokenRights));
-        }
-      }
-    }
-
-    //Either for admin or manage spaces, the packages access must be tested
-    if (isPackagesEdit) {
-      //Requester must hold manageSpaces permission when adding new packages to space
-      if (task.isCreate() || isPackagesAdded(head, target)) {
-        // to add or edit a package within a space, you must have space owner permissions
-        requestRights.manageSpaces(xyzhubFilter);
-        getPackagesFromInput(entry).forEach(packageId -> requestRights.managePackages(
-            XyzHubAttributeMap.forIdValues(target.getOwner(), packageId)));
-      } else {
-        boolean isOwner = tokenRights != null && tokenRights.matches(new XyzHubActionMatrix().manageSpaces(xyzhubFilter));
-        if (!isOwner) {
-          // then it's just a package removal
-          List<String> removedPackages = new ArrayList<>(getPackagesFromSpace(head));
-          removedPackages.removeAll(getPackagesFromSpace(target));
-          removedPackages.forEach(packageId -> requestRights.managePackages(XyzHubAttributeMap.forIdValues(target.getOwner(), packageId)));
-        }
-      }
-    }
-
-    //Checks if the user has useCapabilities: ['searchablePropertiesConfiguration']
-    if (isSearchablePropertiesEdit || isSortablePropertiesEdit) {
-      requestRights.useCapabilities(new AttributeMap().withValue(ID, "searchablePropertiesConfiguration"));
-    }
-
-    //If this is an edit on admin properties.
-    if (isAdminEdit) {
-      boolean ownerChanged = !task.isCreate() && input.containsKey("owner") && !input.get("owner").equals(head.getOwner());
-      if (ownerChanged) {
-        XyzHubAttributeMap additionalNeededPermission = new XyzHubAttributeMap();
-        additionalNeededPermission.withValue(SPACE, head.getId());
-        additionalNeededPermission.withValue(OWNER, input.get("owner"));
-        requestRights.adminSpaces(additionalNeededPermission);
-      }
-
-      requestRights.adminSpaces(xyzhubFilter);
-    }
-
-    if (isBasicEdit) {
-      requestRights.manageSpaces(xyzhubFilter);
+      if (isProcessorsEdit) connectorIds.addAll(getConnectorIds(input, PROCESSORS));
+
+      connectorIds.forEach(id -> {
+        CompletableFuture<Void> f = new CompletableFuture();
+        futureList.add(f);
+        Service.connectorConfigClient.get(task.context.get("marker"), id, ar -> {
+          if (ar.succeeded()) {
+            Connector c = ar.result();
+            connectorsRights.accessConnectors(XyzHubAttributeMap.forIdValues(c.owner, c.id));
+            f.complete(null);
+          } else {
+            //If connector does not exist.
+            connectorsRights.accessConnectors(XyzHubAttributeMap.forIdValues(id));
+            f.complete(null);
```

Comment:
Please log the issue found when "ar" is not succeeded.

Reply:
Added logging of the cause of not getting the connector.